### ISSUE TITLE
Latex in a plotly figure is not rendered

### ISSUE THREAD

**Issue by Midnighter** [spatialaudio/nbsphinx#369]
I'm working on a project that includes some notebooks in the documentation. We have both plots created with matplotlib and plotly in those. The axis titles of those plots contain latex in order to render units nicely. At the moment the matplotlib ones render properly whereas the plotly latex is not rendered in the documentation. Both render perfectly fine when used interactively.

You can reproduce this issue by checking out the [corresponding branch](https://gitlab.com/davidtourigny/dynamic-fba/tree/refactor-plot) installing the dependencies and generating the documentation.

I'm currently using the following versions:

```
nbsphinx==0.5.0
Sphinx==2.3.1
```

**Comment by mgeier** [comment]
Can you please point me to an example of those axis titles?

**Comment by Midnighter** [comment]
Once you have built the docs you can find an example at: `docs/_build/html/example1.html#7.-Plotting-the-results`.

**Comment by mgeier** [comment]
BTW, you should use the Matplotlib settings recommended at https://nbsphinx.readthedocs.io/en/latest/code-cells.html#Plots

**Comment by mgeier** [comment]
I don't know what I'm supposed to see there.

Can you please describe what you are expecting to see there and what you are actually seeing?

Probably some screenshots would help clarify?

**Comment by Midnighter** [comment]
So what I'm expecting is axis titles that look like the following from matplotlib:

![matplotlib](https://user-images.githubusercontent.com/135653/71591239-e8f24100-2b2b-11ea-85ba-e260e8892469.png)

What I'm seeing instead with plotly is the raw latex code:

![plotly](https://user-images.githubusercontent.com/135653/71591273-09220000-2b2c-11ea-9a8d-990c67149e8e.png)

**Comment by mgeier** [comment]
OK, thanks, with those screenshots your description makes more sense ...

I don't see any of this LaTeX stuff in the axis labels ... are you sure you have pushed all your changes to your branch?

**Comment by Midnighter** [comment]
None of that is on readthedocs yet if that is what you mean. Everything is pushed to the branch, though, so if you build it locally you should definitely arrive at the same result.

**Comment by mgeier** [comment]
OK, sorry, I've deleted the cell outputs and re-created them and the LaTeX stuff was gone.

If I use your pre-generated outputs, I see the LaTeX problem ...

**Comment by mgeier** [comment]
In your generated HTML files some MathJax-related stuff seems to be missing.

I would expect the HTML to contain something like this:

```html
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
```

The first tag is normally added by Sphinx when `sphinx.ext.mathjax` is used.

The second tag is normally added by `nbsphinx`, see

https://github.com/spatialaudio/nbsphinx/blob/7a6d64fa1a83e36e23ff27570fdac1f96988748a/src/nbsphinx.py#L1825-L1835

I had no clue why those are missing in your case ... but then it dawned on me:

Sphinx has this feature where MathJax is only loaded if at least one equation is used on some page.
Since you apparently are not using any equations (that Sphinx is aware of), it is not loaded!

The solution is to add some equation somewhere and everything works!

Maybe there is another way to get Sphinx to load MathJax ... I don't know. You could ask the Sphinx folks.

**Comment by Midnighter** [comment]
Indeed, that fixes it! Thank you very much. I will see what I can find out about always including mathjax.

**Comment by mgeier** [comment]
> Maybe there is another way to get Sphinx to load MathJax ... I don't know.

Now I know: #551.
